segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -192,66 +168,35 @@ public void setupAppend(SetupAppend setupAppend) {
     }
 
     /**
-     * If there isn't already an append outstanding against the store, write a new one.
-     * Appends are opportunistically batched here. i.e. If many are waiting they are combined into a single append and
-     * that is written.
+     * Append data to the store.
      */
-    private void performNextWrite() {
-        Append append = getNextAppend();
-        if (append == null) {
-            return;
-        }
-        long traceId = LoggerHelpers.traceEnter(log, "storeAppend", append);
+    @Override
+    public void append(Append append) {
+        long traceId = LoggerHelpers.traceEnter(log, "append", append);
+        UUID id = append.getWriterId();
+        WriterState state = this.writerStates.get(Pair.of(append.getSegment(), id));
+        Preconditions.checkState(state != null, "Data from unexpected connection: %s.", id);
+        long previousEventNumber = state.beginAppend(append.getEventNumber());
+        int appendLength = append.getData().readableBytes();
+        adjustOutstandingBytes(appendLength);
         Timer timer = new Timer();
-        storeAppend(append)
-                .whenComplete((length, e) -> {
-                    handleAppendResult(append, length, e, timer);
-                    LoggerHelpers.traceLeave(log, "storeAppend", traceId, append, e);
+        storeAppend(append, previousEventNumber)
```

Comment:
To confirm the obvious: the store guarantees that the order of calls to append will be guaranteed in the event that another append is passed before the last one has completed.

Reply:
Yes. There is a great deal of effort in the `StreamSegmentContainer.java` and `MetadataStore.java` that ensures that, even if intermediate async calls are ack-ed out of order.